src/JustEat.StatsD/PooledUdpTransport.cs

```diff
@@ -84,12 +72,11 @@ private void Dispose(bool disposing)
         {
             if (!_disposed)
             {
-                Socket socket = null;
-
                 try
                 {
-                    while ((socket = _socketPool.Pop()) != null)
+                    while (_socketPool.Count > 0)
                     {
+                        var socket = _socketPool.Pop();
```

Comment:
The `null`-check has gone. I haven't read the diff of the pool itself yet, but if it can return null we should check so we don't have the potential for a race-condition where between checking the size and calling `Pop()` the pool has emptied.